code/game/machinery/flasher.dm

```diff
@@ -97,24 +100,23 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/flasher, 26)
 	if (!powered() || !bulb)
 		return
 
-	if (bulb.burnt_out || (last_flash && world.time < src.last_flash + 150))
+	if (bulb.burnt_out || !COOLDOWN_FINISHED(src, flash_cooldown))
 		return
 
 	if(!bulb.flash_recharge(30)) //Bulb can burn out if it's used too often too fast
 		power_change()
 		return
 
-	playsound(src.loc, 'sound/weapons/flash.ogg', 100, TRUE)
+	playsound(src, 'sound/weapons/flash.ogg', 100, TRUE)
 	flick("[base_icon_state]_flash", src)
-	set_light_on(TRUE)
-	addtimer(CALLBACK(src, .proc/flash_end), FLASH_LIGHT_DURATION, TIMER_OVERRIDE|TIMER_UNIQUE)
+	flash_lighting_fx()
 
-	last_flash = world.time
+	COOLDOWN_START(src, flash_cooldown, flash_cooldown_duration)
 	use_power(1000)
 
 	var/flashed = FALSE
 	for (var/mob/living/L in viewers(src, null))
```

Comment:
while youre touching this, viewers is only mobs i think? so you can use as anything, and change the name from L

Reply:
that'd include camera mobs & ghosts.

Reply:
just change the name then